src/components/timers/XY.tsx

```diff
@@ -1,3 +1,123 @@
+
+    useEffect(() => {
+        let interval = null;
+
+        if (isRunning) {
+            if (roundsRemaining === Number(roundAmountInput)) {
+                setRoundsRemaining(roundsRemaining - 1);
+            }
+            if (seconds === 0) {
+                if (roundsRemaining === 0) {
+                    setIsRunning(false);
+                    setIsDone(true);
+                } else {
+                    setRoundsRemaining(roundsRemaining - 1);
+                    setSeconds(userInputCleanup(roundDurationInput));
+                }
+            }
+            interval = setInterval(() => {
+                setSeconds(prevseconds => prevseconds - 1);
+            }, 1000);
+        } else if (!isRunning && seconds !== 0 && interval != null) {
+            clearInterval(interval);
+        }
+        if (interval != null) {
+            return () => clearInterval(interval);
+        }
+    }, [isRunning, seconds, roundsRemaining, roundDurationInput, roundAmountInput]);
+
+    const timeChange = () => {
+        setIsRunning(!isRunning);
+    };
+    const handleReset = () => {
+        setIsDone(false);
+        setSeconds(userInputCleanup(roundDurationInput));
+        setRoundsRemaining(Number(roundAmountInput));
+        setIsRunning(false);
+    };
+
+    const handleInputBtnClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        const target = event.target as HTMLElement;
+        if (target.innerText === 'Set') {
+            setSeconds(userInputCleanup(roundDurationInput));
+            setRoundsRemaining(Number(roundAmountInput));
+            if (userInputCleanup(roundDurationInput) > 0 && Number(roundAmountInput) > 0) {
+                setIsInputSet(true);
+            }
+        } else if (target.innerText === 'Clear') {
+            setRoundDurationInput('0000');
+            setRoundAmountInput('00');
+        } else {
+            if (userSelect === 'Time') {
+                setRoundDurationInput(roundDurationInput.slice(1) + target.innerText);
+            } else if (userSelect === 'Rounds') {
+                setRoundAmountInput(roundAmountInput.slice(1) + target.innerText);
+            }
+        }
+    };
+    const handleBackBtn = () => {
+        setIsRunning(false);
+        setRoundDurationInput('0000');
+        setRoundAmountInput('00');
+        setUserSelect('');
+        setIsInputSet(false);
+        setIsDone(false);
+    };
+    const radioChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const eventValue = event.target.value;
+        setUserSelect(eventValue);
+    };
+    const handleFF = () => {
+        setRoundsRemaining(0);
+        setIsDone(true);
+    };
+
+    return (
+        <div className="clockContainer">
+            <div>
+                {isInputSet ? <p className="supportingText">Rounds Remaining: {roundsRemaining}</p> : <p className="supportingText">Rounds: {roundAmountInput}</p>}
+                {isDone ? (
```

Comment:
Nesting ternary operators is difficult to read. In most cases, readability is more important than compactness or having the most optimized and efficient code. 

This should be re-factored to make it more clear what's happening. When things start to feel busy or complicated, you should break it out into its own component. 

Ex. 

```
                {isInputSet ? <p className="supportingText">Rounds Remaining: {roundsRemaining}</p> : <p className="supportingText">Rounds: {roundAmountInput}</p>}
                {isDone ? (
                    <h1 className="clockStyle">DO:NE</h1>
                ) : isInputSet ? (
                    <TimerDisplay seconds={seconds} />
                ) : (
                    <p className="supportingText">Time per Round: {`${roundDurationInput.slice(0, 2)}:${roundDurationInput.slice(2, 4)}`}</p>
                )}
```
That could all be its own component, that receives the required values as props.